zetasql-toolkit-bigquery/src/main/java/com/google/zetasql/toolkit/catalog/bigquery/BigQueryCatalog.java

```diff
@@ -531,6 +569,23 @@ public void addTables(List<String> tableReferences) {
                     table, CreateMode.CREATE_OR_REPLACE, CreateScope.CREATE_DEFAULT_SCOPE));
   }
 
+  private void addWildcardTables(List<String> tableReference) {
```

Comment:
I wouldn't register all tables matching a wildcard. That could be thousands of tables the catalog will never use.

What we should do for a wildcard table is find a single concrete table matching it, get its schema and register one table with that schema under the wildcard name.

For example; for `bigquery-public-data.noaa_gsod.gsod202*` we should only register a table with that name exact name and the corresponding schema. Compared to the current iteration, which would register `gsod202*`, `gsod2020`, `gsod2021`, `gsod2022` and `gsod2023`.